src/connector/src/source/filesystem/opendal_source/s3_source.rs

```diff
@@ -58,8 +58,6 @@ impl<Src: OpendalSource> OpendalEnumerator<Src> {
             );
         }
 
-        builder.enable_virtual_host_style();
```

Comment:
cherry-pick it into release-1.9 because some users using the S3 bucket name with dot `.` and will meet error here https://github.com/apache/opendal/blame/ac694375da955ea6d3461aee0bfafd5a5d774541/core/src/services/s3/backend.rs#L600-L614